crates/net/network/src/transactions/config.rs

```diff
@@ -149,3 +155,124 @@ impl FromStr for TransactionPropagationKind {
+            }
+        }
+    }
+}
+
+/// An `AnnouncementFilteringPolicy` that is more permissive towards unknown transaction types.
+///
+/// Allows for handling of new or future transaction types that a local node may not yet fully
+/// support or recognize.
+#[derive(Debug, Clone, Default)]
+pub struct RelaxedAnnouncementFilter;
+
+impl AnnouncementFilteringPolicy for RelaxedAnnouncementFilter {
+    fn decide_on_announcement(&self, ty: u8, hash: &B256, size: usize) -> AnnouncementAcceptance {
+        match TxType::try_from(ty) {
+            Ok(_valid_type) => AnnouncementAcceptance::Accept,
+            Err(_) => {
+                tracing::trace!(target: "net::tx::policy::relaxed",
+                    %ty,
+                    %size,
+                    %hash,
+                    "Unknown transaction type in announcement. Ignoring entry."
+                );
+                AnnouncementAcceptance::Ignore
+            }
+        }
+    }
+}
+
+/// A container for policies that govern the behaviour of the `TransactionManager`.
+///
+/// This struct bundles different policies, such as how transactions are propagated and how incoming
+/// transaction announcements are filtered.
+#[derive(Debug, Clone)]
+pub struct NetworkPolicies<P, A>
+where
+    P: TransactionPropagationPolicy,
+    A: AnnouncementFilteringPolicy,
+{
+    /// Policy determining to which peers transactions are propagated.
+    pub propagation: P,
+    /// Policy determining how incoming transaction announcements are filtered,
+    /// especially concerning transaction types.
+    pub announcement_filter: A,
+}
+
+impl<P, A> NetworkPolicies<P, A>
+where
+    P: TransactionPropagationPolicy,
+    A: AnnouncementFilteringPolicy,
+{
+    /// Creates a new `NetworkPolicies` instance.
+    pub const fn new(propagation: P, announcement_filter: A) -> Self {
+        Self { propagation, announcement_filter }
+    }
+
+    /// Returns a reference to the configured transaction propagation policy.
+    pub const fn propagation_policy(&self) -> &P {
+        &self.propagation
+    }
+
+    /// Returns a reference to the configured announcement filtering policy.
+    pub const fn announcement_filtering_policy(&self) -> &A {
+        &self.announcement_filter
+    }
+}
+
+/// Specifies the kind of transaction announcement filtering policy to be used by the
+/// `TransactionManager`.
+///
+/// This enum is typically set in the `TransactionManagerConfig` to allow users to choose how
+/// strictly the node should validate announced transactions.
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum AnnouncementFilterKind {
+    /// Use the `StrictAnnouncementFilter`, which rejects and penalizes unknown transaction types.
+    #[default]
+    Strict,
+    /// Use the `RelaxedAnnouncementFilter`, which ignores unknown transaction types.
+    Relaxed,
+}
```

Comment:
do we even need this as part of the config?

Reply:
Not really no, removed.